fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/StandardIndexMaintainer.java

```diff
@@ -365,6 +368,48 @@ public void updateUniquenessViolations(@Nonnull Tuple valueKey, @Nonnull Tuple p
         return keyValues.map(kv -> unpackKeyValue(uniquenessViolationsSubspace, kv));
     }
 
+    /**
+     * Validate the integrity of the index (such as identifying index entries that do not point to records or
+     * identifying records that do not point to valid index entries). The default implementation provided by the
+     * <code>StandardIndexMaintainer</code> class is a no-op (performs no validation) and should be overridden by
+     * implementing classes.
+     * @param continuation any continuation from a previous validation invocation
+     * @param scanProperties skip, limit and other properties of the validation (use default values if <code>null</code>)
+     * @return an empty cursor
```

Comment:
Oh, I realized after this that "an empty cursor" might not be right if the extending subclass has overridden this method. Maybe saying that it returns a cursor over any invalid index entries with their associated reason? I'm not sure.

Reply:
I agree but I think we should also include what this method really returns. What about:
```
@return a cursor over invalid index entries including reasons (the default is an empty cursor)
```

Reply:
Yeah, that sounds good to me